sensors:
- name: kinect
  type: kinect
  source: /camera/rgbd #launch rgbd rose_server.launch for this to be provided
  frame_id: /camera_rgb_optical_frame
  visualize: 0
  voxel_size: 0.02
  max_range: 2.0
  clearing_padding_fraction: 0.05
  normal_k_search: 15
  association_modules:
  - type: PointNormal
    parameters:
        association_correspondence_distance: 0.3
        position_weight: 2.0
        normal_weight: 1.0
        render_width: 320
        render_max_range: 2.2
        render_voxel_size: 0.02
        normal_k_search: 15
        visualize: 0
  - type: PolygonHeight
    parameters:
        max_range: 2.0
        tolerance: 0.035
        min_cluster_size: 20
        visualize: 0
  segmentation_modules:
  - type: EuclideanClustering
    parameters:
        tolerance: 0.035
        min_cluster_size: 20
        visualize: 0

visualize: 1
world_name: rosebv

perception:
    fit_shapes: 1
    model_list: all_models.yml
    # modules:
#    -  lib: libsize_matcher.so
#    -  lib: libed_odu_finder.so
#    -  lib: libcolor_matcher.so
#    -  lib: libhuman_contour_matcher.so
    # -  lib: libface_detector.so
    # -  lib: libtype_aggregator.so

plugins:
- name: gui_server
  lib: libed_gui_server_plugin.so
  parameters:
      robot_name: rose
# - name: builder
#   lib: libed_builder_plugin.so
#- name: obstacle_grid_publisher
#  lib: libed_occupancy_grid_publisher_plugin.so
#  parameters:
#      frequency: 10
#      resolution: 0.05
#      topic: /map_planner
#      frame_id: map
#      sim_time: 1.0
#      object_persistence_time: 10
#- name: loc_grid_publisher
#  lib: libed_occupancy_grid_publisher_plugin.so
#  parameters:
#      frequency: 1
#      resolution: 0.025
#      topic: map
#      frame_id: map
#      specifier: localizable
#      sim_time: 0.0
- name: navigation
  lib: libed_navigation_plugin.so
  parameters:
      occupancy_grid_publisher:
          resolution: 0.05
          frame_id: map
          min_z: 0.05
          max_z: 1.8
- name: tf_publisher
  lib: libed_tf_publisher_plugin.so
  parameters:
      root_frame_id: map
#- name: amigo_plugin
#  lib: libed_robot_plugin.so
#  parameters:
#      frequency: 100
#      robot_name: amigo
#      urdf_rosparam: /amigo/robot_description
#      joint_topics:
#          - topic: /amigo/torso/measurements
#          - topic: /amigo/left_arm/measurements
#          - topic: /amigo/right_arm/measurements
#          - topic: /amigo/neck/measurements
- name: localization
  lib: libed_localization_plugin.so
  parameters:
      initial_pose_topic: /initialpose
      num_particles: 500
      initial_pose:
          x: 0
          y: 0
          rz: 0
      laser_model:
          topic: /scan
          num_beams: 100
          z_hit: 0.95
          sigma_hit: 0.2
          z_short: 0.1
          z_max: 0.05
          z_rand: 0.05
          lambda_short: 0.1
          range_max: 10
          min_particle_distance: 0.01
          min_particle_rotation_distance: 0.02
      odom_model:
          map_frame: map
          odom_frame: /odom
          base_link_frame: /base_link
          alpha1: 2.2    # rot -> trans + strafe
          alpha2: 2.8    # trans -> rot
          alpha3: 2.8    # trans -> trans
          alpha4: 2.4    # rot -> rot
          alpha5: 0.2    # trans -> strafe
